cdap-common/src/main/java/co/cask/cdap/common/UnauthorizedException.java

```diff
@@ -16,21 +16,39 @@
 
 package co.cask.cdap.common;
 
+import co.cask.cdap.proto.id.EntityId;
+import co.cask.cdap.proto.security.Action;
+import co.cask.cdap.proto.security.Principal;
+
 /**
  * Thrown when a user is not authorized to perform an operation.
  */
 public class UnauthorizedException extends Exception {
 
+  private final boolean authenticationError;
+
   public UnauthorizedException() {
     super();
+    this.authenticationError = true;
   }
 
   public UnauthorizedException(String msg, Throwable throwable) {
     super(msg, throwable);
+    this.authenticationError = true;
   }
 
   public UnauthorizedException(String message) {
     super(message);
+    this.authenticationError = true;
```

Comment:
This seems to be called from place where response code is HTTP_UNAUTHORIZED so why set authenticationError true ?
Also what does it exactly means in the context of Authorization ?